### ISSUE TITLE
bundler issues.

### ISSUE THREAD

**Issue by finetype** [MithrilJS/mithril.js#2136]
## Expected Behavior
Mithril's bundler shouldn't produce syntax errors that fail.

## Current Behavior
running the bundler as described on the installation page, I get a syntax error in the output, where it adds
```
;(function(){
//;(function(){
  <code here for most of app>
})()
<more code here from, e.g., "model.js">
})()
```

## Possible Solution
Not sure how the bundler works, sorry. :(

## Steps to Reproduce (for bugs)

This happens when I have
`// var m = require("mithril")`
at the top of "model.js"

If I remove that line, all is well. If I leave that line there uncommented, all is well--technically, that is. It just doesn't comment out that "second" scope block, so no syntax error. Still awkwardly doubles up on its scope stuff.

loading pattern:
`index.html` contains
-> `<script src=index.js>`
`index.js` contains
-> `var Model = require("./model")`

## Context
Just trying to build an alpha mvp of a concept. I built out the part that accesses a google sheet with data and loads it into memory. Now I'm trying to build out a dynamic front end to play with.

## Your Environment
it's pretty bare bones.
* Version used: using CDN from docs @ "http://unpkg.com/mithril/mithril.js"
* Browser Name and version: Chromium Version 64.0.3282.119 (Developer Build) built on Debian 9.3, running on Debian 9.2 (64-bit)
* Operating System and version (desktop or mobile): debian 9 desktop
* Link to your project: https://gitlab.com/finetype/globetrotters-phrasebook <-- the code that produced the error has since been changed significantly, in part to fix this error. I was moving very quickly, so I don't recall how I fixed the problem in the end.

PS--also noticed weirdness around how it handled the "require" statement... "require("model.js") got bundler errors like "cannot find model.js.js". just having "model.js" didn't work, either, even though it's in the same directory level. 

Other problems... mithril bundler renamed one of my object keys in one file because a variable with that name occurred in another file. That caused other code within that file to fail, however, since I get text from an xhr call that needs to match that property name exactly. I renamed a variable to fix this, but it was appending a "0" to one var name, breaking stuff.

also it stopped watching reliably, then seemingly at all? Even though I was calling `npm run-script build`, with `"build": "bundle client/index.js --output bin/app.js --watch"` in package.json.

**Comment by pygy** [comment]
As you've realized, the bundler is pretty limited in what it can do. It uses regexps to parse the source, which leads to a host of issues.

Your main issue (a commented out require causing havoc) could be solved by roughly lexing the source using regexps to ignore comments (that can be done cleanly).

Solving the variable alias thing would require a full JS parser. I know that @lhorie has been working on one, but I don't think that even he would push the zero-dependencies thing as far as having a JS parser included in this repo ;-)

Not sure about the `.js.js` thing.

More generally, it isn't advisable to use the bundler for things other than Mithril itself.

**Comment by finetype** [comment]
No worries, I understand. I was using this as a shortcut to work an mvp, and read the caveats in advance. In the gitter chat, when I asked whether bug reports on it were welcome, I was encouraged to post, so I did so.

Seems like reinventing the wheel to write a parser by hand.

To be fair, though, the install page explicitly offers the bundler as a viable option. It says it isn't production ready, but it implies that it is not unreasonable to use otherwise. It doesn't mention that it should not be used for anything other than Mithril itself. Not sure if that page should be updated? Might turn someone off of mithril if they try that route and find a bunch of bugs almost immediately--they might infer a lack of reliability in the client code, which would be unfortunate.

**Comment by tivac** [comment]
We should deprecate the bundler and move to `rollup`, but that's a longer-term project.

Appreciate the bug report but I've been in the guts of the bundler and I'm not very interested in going back. If you wanted to send a PR I'd certainly take a look at it!

**Comment by finetype** [comment]
Nah, I support the move to an existing tool. I get and appreciate minimizing dependencies, but writing a custom bundler for development  to save a couple kb's seems like overkill. Cheers.

**Comment by StephanHoyer** [comment]
we will kick the bundler, so won't fix
